### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by kalimullina**

### CODE REVIEW

**Comment by ZeroShadows** [comment]
general comment 
lets create abstract base test class and move test setup, test tear down and driver initialisation there

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class Ex1 {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: rename test class with specific name

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    @BeforeClass
+    public static void setupClass() {
+        WebDriverManager.chromedriver().setup();
+    }
+
+    @Test
+    public void userHomePageTest() {
+        webDriver = new ChromeDriver();
+        webDriver.manage().window().maximize();
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: move it to test setup

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class Ex1 {
+    private WebDriver webDriver;
+    private WebElement webElement;
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: why do you need it as class field?

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        webDriver.findElement(By.className("icon-practise")).isDisplayed();
+        webDriver.findElement(By.className("icon-custom")).isDisplayed();
+        webDriver.findElement(By.className("icon-multi")).isDisplayed();
+        webDriver.findElement(By.className("icon-base")).isDisplayed();
+
+
+
+        //7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        webElement = webDriver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div[2]/div[1]/div/span"));
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: you could use var variableName for that.
From my point of view it's a bit strange to use one WebElement variable for the whole test

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        assertThat(webElement.getText())
+            .as("Fisrt text on the Index Page under icon was wrong")
+            .isEqualTo("To include good practices\nand ideas from successful\nEPAM project");
+        webElement = webDriver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div[2]/div[2]/div/span"));
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: same for variable here

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        assertThat(webElement.getText())
+            .as("Second text on the Index Page under icon was wrong")
+            .isEqualTo("To be flexible and\ncustomizable");
+        webElement = webDriver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div[2]/div[3]/div/span"));
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: same for variable here

File: src/test/java/com/epam/tc/hw2/Ex1.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        assertThat(webElement.getText())
+            .as("Third text on the Index Page under icon was wrong")
+            .isEqualTo("To be multiplatform");
+        webElement = webDriver.findElement(By.xpath("/html/body/div/div[2]/main/div[2]/div[2]/div[4]/div/span"));
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex1.java]: same for variable here

File: src/test/java/com/epam/tc/hw2/Ex2.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class Ex2 {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw2/Ex2.java]: same comments as in Ex1

File: src/test/resources/hw2/Ex1Suite.xml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+
+<suite name="homework2_Ex1_Suite">
+  <test name="homework2_Ex1_Test">
+    <classes>
+      <class name="com.epam.tc.hw2.Ex1"/>
```
Comment (ZeroShadows) [inline src/test/resources/hw2/Ex1Suite.xml]: add second test class here and remove Ex2.xml

**Review by ZeroShadows — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 1 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 2 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 3 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: webElement made to local variable

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 4 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 5 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 6 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex1.java (hunk 7 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex1.java]: fixed

File: src/test/java/com/epam/tc/hw2/Ex2.java (hunk 8 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw2/Ex2.java]: fixed

File: src/test/resources/hw2/Ex1Suite.xml (hunk 9 above)
Comment (kalimullina) [reply src/test/resources/hw2/Ex1Suite.xml]: fixed

**Review by ZeroShadows — APPROVED** [review]

**Pull request merged**